### PR TITLE
Lc 136 eventhubs get output input topics from kcql

### PR DESCRIPTION
**Pull request by GoMati-MU**
#### Description
Initial implementation for basic KCQL support in the form of e.g. `INSERT INTO output SELECT * FROM input`. This should more-or-less close the MVP of Azure Event Hubs.

#### Testing
Additional unit tests added around KCQL, changes to some that were already checking for late `topics` config key.

### CODE REVIEW

File: java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java [hunk 1]
```diff
@@ -87,4 +94,21 @@ public void stop() {
...
+   * @param azureEventHubsSourceConfig task configuration
+   * @return output topics list
+   */
+  private List<String> getOutputTopicsFromConfig(
+      AzureEventHubsSourceConfig azureEventHubsSourceConfig) {
+    Kcql kcql = KcqlConfigPort.parseMultipleKcqlStatementsPickingOnlyFirst(
+        azureEventHubsSourceConfig.getString(AzureEventHubsConfigConstants.KCQL_CONFIG));
+    List<String> outputTopics = Arrays.stream(kcql.getTarget().split(",")).collect(Collectors.toList());
+    if (outputTopics.isEmpty()) {
+      throw ExceptionProviders.OUTPUT_TOPIC_CONFIG_EXCEPTION_SUPPLIER.get();
```
Comment (stheppi) [inline java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java:110]: i don't think throwing exception for empty is adding value.

File: java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java [hunk 2]
```diff
@@ -87,4 +94,21 @@ public void stop() {
...
+   * Returns output topics (if specified in config) or throws {@link ConfigException}.
+   *
+   * @param azureEventHubsSourceConfig task configuration
+   * @return output topics list
+   */
+  private List<String> getOutputTopicsFromConfig(
+      AzureEventHubsSourceConfig azureEventHubsSourceConfig) {
+    Kcql kcql = KcqlConfigPort.parseMultipleKcqlStatementsPickingOnlyFirst(
+        azureEventHubsSourceConfig.getString(AzureEventHubsConfigConstants.KCQL_CONFIG));
+    List<String> outputTopics = Arrays.stream(kcql.getTarget().split(",")).collect(Collectors.toList());
```
Comment (stheppi) [inline java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java:108]: This is confusing:

Consider these kcql statements
```
INSERT into a,b,c SELECT * FROM X;
```
This is not valid SQL. Sure KCQL is not SQL92 even, but why would you build such a functionality?

File: java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/BlockingQueueProducerProvider.java [hunk 3]
```diff
@@ -78,4 +83,21 @@ private boolean shouldConsumerSeekToLatest(AzureEventHubsConfig azureEventHubsCo
...
+   * Returns input topics (if specified in config) or throws {@link ConfigException}.
+   *
+   * @param azureEventHubsSourceConfig task configuration
+   * @return input topics list
+   */
+  private String getInputTopicFromConfig(
+      AzureEventHubsSourceConfig azureEventHubsSourceConfig) {
+    Kcql kcql = KcqlConfigPort.parseMultipleKcqlStatementsPickingOnlyFirst(
+        azureEventHubsSourceConfig.getString(AzureEventHubsConfigConstants.KCQL_CONFIG));
+    List<String> inputTopics = Arrays.stream(kcql.getSource().split(",")).collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (GoMati-MU) [reply java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/BlockingQueueProducerProvider.java]: I removed this code, thanks!

File: java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java (hunk 2 above)
Comment (GoMati-MU) [reply java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java]: I removed this code, thanks!

File: java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java (hunk 1 above)
Comment (GoMati-MU) [reply java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java]: Removed, I'll just leave it to KCQL parsing then.

**Comment by GoMati-MU** [comment]
Comments up to this point have been addressed. There is a possibility for a change of requirements but it has to be first decided upon.

**Review by stheppi — APPROVED** [review]

**Pull request merged**